courier/template/load_template_test.go

```diff
@@ -61,4 +66,83 @@ func TestLoadTextTemplate(t *testing.T) {
 		require.NoError(t, os.RemoveAll(fp))
 		assert.Contains(t, executeTextTemplate(t, dir, name, "", nil), "cached stub body")
 	})
+
+	t.Run("method=remote resource", func(t *testing.T) {
+		t.Run("case=base64 encoded data", func(t *testing.T) {
+			t.Run("html template", func(t *testing.T) {
+				m := map[string]interface{}{"lang": "en_US"}
+				f, err := ioutil.ReadFile("courier/builtin/templates/test_stub/email.body.html.nested.gotmpl")
+				require.NoError(t, err)
+				b64 := base64.StdEncoding.EncodeToString(f)
+				tp, err := LoadHTMLTemplate(nil, "", "", m, "base64://"+b64, "base")
+				require.NoError(t, err)
+				assert.Contains(t, tp, "lang=en_US")
+			})
+
+			t.Run("case=plaintext", func(t *testing.T) {
+				m := map[string]interface{}{"Body": "something"}
+				f, err := ioutil.ReadFile("courier/builtin/templates/test_stub/email.body.plaintext.gotmpl")
+				require.NoError(t, err)
+
+				b64 := base64.StdEncoding.EncodeToString(f)
+
+				tp, err := LoadTextTemplate(nil, "", "", m,
+					"base64://"+b64, "base")
+				require.NoError(t, err)
+				assert.Contains(t, tp, "stub email body something")
+			})
+
+		})
+
+		t.Run("case=file resource", func(t *testing.T) {
+			t.Run("case=html template", func(t *testing.T) {
+				m := map[string]interface{}{"lang": "en_US"}
+				tp, err := LoadHTMLTemplate(nil, "", "", m,
+					"file://courier/builtin/templates/test_stub/email.body.html.nested.gotmpl",
+					"base",
+				)
+				require.NoError(t, err)
+				assert.Contains(t, tp, "lang=en_US")
+			})
+
+			t.Run("case=plaintext", func(t *testing.T) {
+				m := map[string]interface{}{"Body": "something"}
+				tp, err := LoadTextTemplate(nil, "", "", m,
+					"file://courier/builtin/templates/test_stub/email.body.plaintext.gotmpl",
+					"base")
+				require.NoError(t, err)
+				assert.Contains(t, tp, "stub email body something")
+			})
+		})
+
+		t.Run("case=http resource", func(t *testing.T) {
```

Comment:
Can you please add a test where the config is set so that `clients.http.disallow_private_ip_ranges=true` and then use a localhost URL and see whether it fails?